torch/csrc/api/include/torch/optim/optimizer.h

```diff
@@ -81,58 +81,47 @@ class TORCH_API OptimizerParamGroup {
   std::unique_ptr<OptimizerOptions> options_;
 };
 
-namespace detail {
-
-/// Base class for all optimizers, that does not yet define a `step()`
-/// mechanism. All it specifies is that optimizers must be supplied with a
-/// vector of parameters. It also defines certain methods that all optimizers
-/// shall have, such as `zero_grad`.
-class TORCH_API OptimizerBase {
+class TORCH_API Optimizer {
  public:
   // The copy constructor is deleted, because the user should use the
   // `state_dict` / `load_state_dict` API to copy an optimizer instead.
-  OptimizerBase(const OptimizerBase& optimizer_base) = delete;
-  OptimizerBase(OptimizerBase&& optimizer_base) = default;
+  Optimizer(const Optimizer& optimizer) = delete;
+  Optimizer(Optimizer&& optimizer) = default;
 
-  /// Constructs the `Optimizer` from a vector of parameters.
-  explicit OptimizerBase(std::vector<Tensor> parameters);
+  Optimizer() : defaults_(nullptr) {}
```

Comment:
I believe we can remove this constructor now, is that correct?

Reply:
yup :D